### PR TITLE
Added scenario [SORDEV-6612] to Case.feature

### PR DESCRIPTION
**Pull request by mk-sgent**
SORQA-94

**Jenkins job:**
**First**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/526/allure/
**Second**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/527/allure/
**third**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/528/allure/
**Fourth**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/530/allure/
**Fifth**
https://jenkins.sormas.netzlink.com/job/sormas-RunE2eTests-branch/682/

### CODE REVIEW

**Comment by rdutu-vg** [comment]
please provide jenkins execution, no more screenshots from local run.

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java [hunk 1]
```diff
@@ -92,6 +93,32 @@ public EditCaseSteps(
...
+        });
+
+    When(
+        "I change Epidemiological confirmation Combobox to {string} option",
+        (String option) -> {
+          webDriverHelpers.selectFromCombobox(EPIDEMIOLOGICAL_CONFIRMATION_COMBOBOX, option);
+        });
+
+    When(
+        "I check that Case Classification has {string} value",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/EditCaseSteps.java:111]: isn't this method the same with existing one?
"For the current Case the Case Classification value should be {string}"

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/test/resources/features/sanity/web/Case.feature [hunk 2]
```diff
@@ -399,3 +399,96 @@ Feature: Case end to end tests
     And I navigate to the last created through API Event page via URL
     And I check that number of displayed Event Participants is 1
 
+  @issue=SORDEV-6612 @env_main
+  Scenario: Manually triggered calculation of case classification
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/resources/features/sanity/web/Case.feature:403]: we already have a class with existing similar test:
Feature: Case Classification functionality

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java [hunk 3]
```diff
@@ -86,6 +87,45 @@ public CreateNewSampleSteps(
           fillCommentsOnSample(sample.getCommentsOnSample());
         });
 
+    When(
+        "^I create a new Sample with for COVID alternative purpose$",
```
[the remark this answers is not in the document]
Comment (mk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java:91]: This one is specific for Covid purpose case and no one methods had before option for thick "Request pathogen tests to be performed"

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/SymptomsTabSteps.java [hunk 4]
```diff
@@ -182,6 +291,57 @@ private Symptoms collectSymptomsData() {
         .build();
   }
 
+  private Symptoms collectSymptomsDataForNoOption() {
```
[the remark this answers is not in the document]
Comment (mk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/SymptomsTabSteps.java:294]: I think method collects symptoms has matter what option was selected before. If we fill Symptoms form for "NO" or "UNKNOWN" options the fields like FIRST SYMPTOM or DATE OF SYMPTOM ONSET are unavailable the validation and more generic validation will be crash.
I know before was existing ""I check the created data is correctly displayed on Symptoms tab page" methods but for NO and UNKNOWN options is crash so I create New specific for No and UNKNOWN option

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/SymptomService.java [hunk 5]
```diff
@@ -60,4 +62,93 @@ public Symptoms buildEditGeneratedSymptoms() {
...
+        .diarrhea(NO.toString())
+        .nausea(NO.toString())
+        .lossOfSmell(NO.toString())
+        .lossOfTaste(NO.toString())
+        .otherNonHemorrhagicSymptoms(NO.toString())
+        .symptomsComments(UUID.randomUUID().toString())
+        .build();
+  }
+
+  public Symptoms buildEditGeneratedSymptomsWithUnknownOptions() {
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/SymptomService.java:126]: create one method for this one and buildEditGeneratedSymptomsWithNoOptions(), and just pass the value which should be set to all fields. Its duplicated code.

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/cases/CaseDirectoryPage.java [hunk 6]
```diff
@@ -162,4 +162,9 @@ public static final By getCaseResultsUuidLocator(String uuid) {
   public static By getResultByIndex(String rowNumber) {
     return By.xpath(String.format("//tr[%s]//a", rowNumber));
   }
+
+  public static final By CASE_CLOSE_WINDOW_BUTTON =
+      By.xpath("//div[contains(@class,\"v-window-closebox\")]");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/cases/CaseDirectoryPage.java:167]: please remove the \ \ from selector

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 7]
```diff
@@ -78,6 +78,15 @@
   public static final By PATHOGEN_LABORATORY_INPUT =
       By.cssSelector(
           "[location='lab'] [class='v-filterselect v-widget v-required v-filterselect-required v-has-width'] input");
+  public static final By REQUEST_PATHOGEN_OPTION_BUTTON =
+      By.cssSelector("[id='pathogenTestingRequested'] label");
+  public static final By ANTIGEN_DETECTION_TEST_OPTION_BUTTON =
+      By.xpath("//label[text()='Antibody detection']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:84]: any other way to take the locator without visible text to make it reusable also for DE?

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 8]
```diff
@@ -78,6 +78,15 @@
...
+  public static final By ISOLATION_TEST_OPTION_BUTTON = By.xpath("//label[text()='Isolation']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:85]: same as above

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java [hunk 9]
```diff
@@ -78,6 +78,15 @@
...
+  public static final By PCR_RTP_PCR_TEST_OPTION_BUTTON =
+      By.xpath("//label[text()='PCR / RT-PCR']");
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:87]: same as above

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/SampleService.java [hunk 10]
```diff
@@ -80,6 +80,21 @@ public Sample buildGeneratedSampleTestResult() {
         .build();
   }
 
+  public Sample buildGeneratedSampleTestResultForCovid() {
+    long currentTimeMillis = System.currentTimeMillis();
+    return Sample.builder()
+        .sampleTestResults("Positive")
+        .reportDate(LocalDate.now().minusDays(10))
+        .typeOfTest("PCR / RT-PCR")
+        .testedDisease("COVID-19")
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/SampleService.java:89]: take the value from enums class

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/SymptomsTabSteps.java [hunk 11]
```diff
@@ -182,6 +297,57 @@ private Symptoms collectSymptomsData() {
...
+            webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(FAST_HEART_RATE_OPTIONS))
+        .diarrhea(webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(DIARRHEA_OPTIONS))
+        .nausea(webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(NAUSEA_OPTIONS))
+        .lossOfSmell(
+            webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(LOSS_OF_SMELL_OPTIONS))
+        .lossOfTaste(
+            webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(LOSS_OF_TASTE_OPTIONS))
+        //        .otherNonHemorrhagicSymptoms(
+        //            webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(
+        //                OTHER_NON_HEMORRHAGIC_SYMPTOMS_OPTIONS))
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/SymptomsTabSteps.java:346]: why commented code?

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java (hunk 7 above)
Comment (mk-sgent) [reply sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/samples/CreateNewSamplePage.java:84]: I talk about this with team and we claim that option is the most safety and optional for this case. For DE case we simply create in future a new locators

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**